pkg/minikube/command/ssh_runner.go

```diff
@@ -189,6 +196,49 @@ func (s *SSHRunner) Copy(f assets.CopyableFile) error {
 	return g.Wait()
 }
 
+func (s *SSHRunner) fileExistsInVM(f assets.CopyableFile, dst string) bool {
+	sess, err := s.c.NewSession()
+	if err != nil {
+		return false
+	}
+
+	// check if sizes of the two files are the same
+	srcSize := f.GetLength()
+	size := fmt.Sprintf("ls -l %s | cut -d \" \" -f5", dst)
+	out, err := sess.CombinedOutput(size)
+	if err != nil {
+		return false
+	}
+	dstSize, err := strconv.Atoi(strings.Trim(string(out), "\n"))
+	if err != nil {
+		return false
+	}
+	if srcSize != dstSize {
+		return false
+	}
+
+	sess, err = s.c.NewSession()
+	if err != nil {
+		return false
+	}
+	// ensure src file hasn't been modified since dst was copied over
+	srcModTime := f.GetModTime()
+	stat := "stat -c %Y" + fmt.Sprintf(" %s", dst)
+	out, err = sess.CombinedOutput(stat)
+	if err != nil {
+		return false
+	}
+	unix, err := strconv.Atoi(strings.Trim(string(out), "\n"))
+	if err != nil {
+		return false
+	}
+	dstModTime := time.Unix(int64(unix), 0)
+	if err != nil {
+		return false
+	}
+	return srcModTime.Before(dstModTime)
```

Comment:
To be on the safe side, I would transfer whenever the modification times differ, otherwise the content might differ. That assumes we are setting the mod time correctly in `Copy()`.

Reply:
It looks like when we copy over the file the mod time gets set to the time of Copy, not the mod time of the original file. I'll try to fix that in Copy!